File: backend/src/storico/api/schemas/extraction.py
"""Extraction Pydantic schemas for Storico API."""

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict

from storico.domain.entities.extraction import ExtractionStatus
from storico.domain.entities.user_story import UserStoryStatus
from storico.domain.entities.task import TaskStatus


class UserStorySchema(BaseModel):
    """User story info included in extraction response."""

    model_config = ConfigDict(from_attributes=True)

    id: UUID
    title: str
    status: UserStoryStatus
    workspace_id: UUID


class TaskSchema(BaseModel):
    """A single task returned in an extraction response."""

    model_config = ConfigDict(from_attributes=True)

    id: UUID
    summary: str
    description: str
    status: TaskStatus
    order_index: int


class ExtractionResponse(BaseModel):
    """Response body representing an LLM extraction result."""

    model_config = ConfigDict(from_attributes=True)

    id: UUID
    user_story_id: UUID
    model_used: str
    status: ExtractionStatus
    user_story_status: UserStoryStatus
    error_info: str | None = None
    prompt_config: dict | None = None
    raw_response: str
    confidence_score: float | None = None
    created_at: datetime
    completed_at: datetime | None = None
    tasks: list[TaskSchema] = []


class ExtractRequest(BaseModel):
    """Request body for task extraction from a user story."""

    user_story_id: UUID
    model: str | None = None
    temperature: float | None = None
    run_validation: bool = False


class ExtractResponse(BaseModel):
    """Response body for a task extraction request."""

    extraction_id: UUID
    status: ExtractionStatus
    user_story_id: UUID
    message: str

File: frontend/src/types/extraction.ts
/** Extraction job status — for polling compatibility. */
export type ExtractionStatus = 'pending' | 'completed' | 'failed';

export type UserStoryStatus =
  'pending_extraction' | 'extracting' | 'extracted' | 'failed_extraction';

export type TaskStatus = 'backlog' | 'todo' | 'in_progress' | 'review' | 'done';

/** Single task in extraction response. */
export interface ExtractionTask {
  id: string;
  summary: string;
  description: string;
  status: TaskStatus;
  order_index: number;
}

/** Full extraction response from GET /extractions/{id}. */
export interface ExtractionResponse {
  id: string;
  user_story_id: string;
  model_used: string;
  status: ExtractionStatus;
  user_story_status: UserStoryStatus;
  error_info: string | null;
  prompt_config: Record<string, unknown> | null;
  raw_response: string;
  confidence_score: number | null;
  created_at: string;
  completed_at: string | null;
  tasks: ExtractionTask[];
}
